tests/components/fujitsu_hvac/test_coordinator.py

```diff
@@ -0,0 +1,75 @@
+"""Test for the FujitsuHVACCoordinator."""
+from unittest.mock import AsyncMock, Mock
+
+from ayla_iot_unofficial import AylaAuthError
+from ayla_iot_unofficial.fujitsu_hvac import FujitsuHVAC
+import pytest
+
+from homeassistant.components.fujitsu_hvac.coordinator import FujitsuHVACCoordinator
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed
+
+TEST_SERIAL_NUMBER = "testserial123"
+TEST_SERIAL_NUMBER2 = "testserial345"
+
+
+async def test_coordinator_initial_data(hass: HomeAssistant) -> None:
+    """Test that coordinator returns the data we expect after the first refresh."""
+    devicemock = AsyncMock(spec=FujitsuHVAC)
+    devicemock.device_serial_number = TEST_SERIAL_NUMBER
+
+    apimock = AsyncMock()
+    apimock.async_get_devices.return_value = [devicemock]
+
+    coordinator = FujitsuHVACCoordinator(hass, apimock)
+    await coordinator.async_config_entry_first_refresh()
+
+    assert coordinator.data == {TEST_SERIAL_NUMBER: devicemock}
```

Comment:
You should not create coordinator nor entities in the tests itself. The tests are kind off end to end based.

Reply:
So instead of testing we fetch the right data, we should test if the state machine is returning what we expect through the entities

Reply:
Not sure I understand. This is supposed to test the coordinator, I don't really see how to exercise the code paths I added without doing what I am doing here.